cmd/bpf2go/gen/output.go

```diff
@@ -141,22 +140,50 @@ func Generate(args GenerateArgs) error {
 		programs[name] = args.Identifier(name)
 	}
 
-	typeNames := make(map[btf.Type]string)
-	for _, typ := range args.Types {
-		// NB: This also deduplicates types.
-		typeNames[typ] = args.Stem + args.Identifier(typ.TypeName())
+	tn := templateName(args.Stem)
+	reservedNames := map[string]struct{}{
```

Comment:
At first sight, I'd model this as a property to be specified during construction of the GoFormatter. Everytime the formatter generates an identifier (struct or otherwise), it can check uniqueness and push it to the set of seen names. I think it's a good feature to have in the formatter to prevent it from emitting invalid code, it doesn't need to be b2g-specific.

This way, you also avoid having to close over `reservedNames` and `typeByName` to get it into an `ShortEnumIdentifier` function.